sync_transaction_test.go

```
@@ -0,0 +1,173 @@
+	if os.Geteuid() == 0 {
+		t.Skip("read-only directories do not block root")
+	}
+	root := t.TempDir()
+	writeTestFile(t, filepath.Join(root, "gen", "changed.go"), "before")
+	writeTestFile(t, filepath.Join(root, "locked", "gen.go"), "locked-before")
+
+	transaction, err := newSyncTransaction(root, "modules/api")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer transaction.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `transaction.Close` is not checked

(errcheck)